### DISCUSSION TITLE
Table Creation Controls (discard modal update)

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#305] [Maintainer discussions]
The changes required for issue #285 are ready for review. Please take a look at the attached document and follow the [review guidelines](https://wiki.mathesar.org/en/design/process/review-guidelines).

## Document Link
https://wiki.mathesar.org/design/process/documents/design/table-creation-controls-spec

@kgodey @pavish your review is needed as this impacts product design and front-end.
@mathemancer your review is optional as this is a design enhancement. But still, take a look and add your comments.

**Comment by pavish** [comment]
@ghislaineguerin The flow looks good to me. I have a few concerns.

1. Can we make the 'Add new table' option, more prominent? The icon in the sidebar is not very prominent, and also since the table selection could change away from a tree based structure, it makes sense to place it somewhere else.
2. The UI assumes the top database + schema selector is present. It makes sense, since that would eventually be required. But currently that is not implemented. @kgodey for this milestone, I'm going to hardcode the schema as 'public' on the UI. When the db selector is complete, the schema would be based on which one is selected. Is this okay?
3. The start with empty table option would not be possible at the moment. It seems to directly go the table edit view, with a default column called id, but this would mean we'll have to create a table on the db, with that column. The flow for this is not yet well defined, and it's not part of this milestone. We could come back to it later. @kgodey, @mathemancer What are your thoughts on this option?

**Reply by ghislaineguerin** [reply]
@pavish Thanks for the feedback. 
1. Yes, I think we need to make it more prominent, and maybe some additional control to add new objects and that is easier to access.
2. Seems like a good idea.
3. We should assume the empty table is there only for demo purposes.

**Reply by kgodey** [reply]
@ghislaineguerin I don't follow what you mean by "the empty table is there only for demo purposes." Demo for what/who?

**Reply by ghislaineguerin** [reply]
@kgodey sorry, I meant it's just for demonstrating what would happen after a table is created, where the name label can be clicked and edited.

**Comment by kgodey** [comment]
The flow looks good to me as well.

@pavish can you hardcode the schema to the first available, rather than `public` specifically? Please also create an issue in the DB milestone to make the switch to using the open schema.

We can skip the empty table part, we have issues to track the work in milestone 7, e.g. https://github.com/centerofci/mathesar/issues/188

**Reply by pavish** [reply]
@kgodey Created issue #316 for this.

**Reply by kgodey** [reply]
Looks good, thanks!

**Comment by ghislaineguerin** [comment]
@pavish @kgodey 
I have updated the sidebar to have fixed headers so that all section headers are visible (tables and views). I think we might also have additional controls at the bottom to add tables and other objects. If possible it would be good to incorporate sorting controls so that the user can have the last modified tables on top, it will be easier to see their new tables listed on top and work through them, rather than have them reordered by name.

**Reply by kgodey** [reply]
@ghislaineguerin This makes sense but I think sidebar changes are probably better implemented as part of https://github.com/centerofci/mathesar/discussions/301 since this only focuses on the table creation flow.

**Comment by ghislaineguerin** [comment]
@kgodey @pavish 
Review notes have been updated for this spec https://wiki.mathesar.org/en/design/specs/table-creation-controls
